man/fi_shm.7.md

```diff
@@ -168,6 +168,13 @@ The *shm* provider checks for the following environment variables:
   page fault is reported, so that there is valid address translation for the
   remaining addresses in the command. This minimizes DSA page faults. Default
   false
+
+*FI_SHM_USE_XPMEM*
+ : SHM can use SAR, CMA or XPMEM for host memory transfer. If
+   FI_SHM_USE_XPMEM is set to 1, the provider will select XPMEM over CMA if
+   XPMEM is available.  Otherwise, if neither CMA nor XPMEM are available
+   SHM shall default to the SAR protocol. Default 0
+
```

Comment:
Since the param is defined with the core_prov instead of the shm provider would the variable be FI_USE_XPMEM? Also this is missing the second xpmem variable for chunk size

Reply:
Why is this global to libfabric, rather than limited to the provider that implements it?

Reply:
I think it could go either way but the xpmem definition outlined here is something usable for all providers so you can condense all of the enabling into one versus having to enable it for different providers using different variables. I think you could make the same argument about other core_prov variables like FI_MR_CACHE_* or FI_HMEM_* variables for example. Is there a value to separating it out between providers? I don't personally think so.
Realistically shm is the only provider that is going to make use of it so I don't think it really matters. I don't care either way.